### PR TITLE
Fixing errors related to strict templates

### LINKED ISSUE #847: Enable strict templates for angular templates

### PR DESCRIPTION
**Pull request by dafo**
Closes #847 .  

Additional information related to this pull request:

### CODE REVIEW

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 1]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	public topSpeedSummary = CustomTopSpeedSummary;
+	public bnpSummary = CustomBPMSummary;
+	public speedSummary = CustomSpeedSummary;
+	public localData: Athlete[] = [];
+	public isFinished = false;
+	public hasWinner = false;
+	public athleteColumnWidth = '30%';
+	public showOverlay = false;
+	public overlaySettings!: OverlaySettings;
+	// eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:52]: you can remove this warning workaround - this should not be needed now

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 2]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+  
+	get showWinnerOverlay() {
+	  return this.showOverlay && this.hasWinner && !this.isFinished;
+	}
+  
+	get showFinishedOverlay() {
+	  return this.showOverlay && this.isFinished;
+	}
+  
+	get hideAthleteNumber() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:81]: set type to `boolean`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 3]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  return this.windowWidth && this.windowWidth < 960;
+	}
+	get hideBeatsPerMinute() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:84]: set type to `boolean`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 4]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  return (this.windowWidth && this.windowWidth < 860) || !this.live;
+	}
+  
+	constructor(@Inject(IgxOverlayService) public overlayService: IgxOverlayService) { }
+	public ngOnInit() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:89]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 5]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  this.localData = athletesData.slice(0, 30).sort((a, b) => b.TrackProgress - a.TrackProgress);
+	  this.localData.forEach(rec => this.getSpeed(rec));
+	  this.windowWidth = window.innerWidth;
+	  this._timer = setInterval(() => this.ticker(), 1500);
+	  this.overlayService.onClosing.subscribe((event: OverlayClosingEventArgs) => {
+		this.showOverlay = false;
+	  });
+	}
+  
+	public ngAfterViewInit() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:99]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 6]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+		this.grid1
+	  );
+	  this.overlaySettings.modal = true;
+	}
+  
+	public getValue(cell: IgxGridCellComponent): number {
+	  const val = cell.value;
+	  return val;
+	}
+	public ngOnDestroy() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:111]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 7]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  clearInterval(this._timer);
+	}
+  
+	public isTop3(cell: IgxGridCellComponent): boolean {
+	  const top = this.grid1.page === 0 && cell.row.index < 4;
+	  return top;
+	}
+  
+	public getTrophyUrl(index: number) {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:120]: set type to `string`. You should return empty string in the end of the method.

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 8]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+		default:
+		  return 'success';
+	  }
+	}
+  
+	public getSpeed(athlete: Athlete): void {
+	  athlete['Speed'] = this.addSpeedeData(athlete, 40);
+	}
+  
+	public getSpeedeData(minutes?: number): any[] {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:160]: after adding `SpeedDescriptor` this should return `SpeedDescriptor[]`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 9]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  if (minutes === undefined) {
+		minutes = 20;
+	  }
+	  const speed: any[] = [];
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:164]: set this to `SpeedDescriptor[]`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 10]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  for (let m = 0; m < minutes; m += 3) {
+		const value = this.getRandomNumber(17, 20);
+		speed.push({ Speed: value, Minute: m });
+	  }
+	  return speed;
+	}
+  
+	public addSpeedeData(athlete: Athlete, minutes?: number): any[] {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:172]: this should return `SpeedDescriptor[]`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 11]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+		if (speedCollection.length === 40) {
+		  speedCollection.shift();
+		}
+	  }
+	  return speedCollection;
+	}
+  
+	// eslint-disable-next-line @typescript-eslint/member-ordering
+	@HostListener('window:resize', ['$event'])
+	public onResize(event: any) {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:192]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 12]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  this.windowWidth = event.target.innerWidth;
+	}
+  
+	public filter(event: Event) {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:196]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 13]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  const target = event.target as HTMLInputElement;
+	  const term = target.value;
+	  this.grid1.filter('CountryName', term, IgxStringFilteringOperand.instance().condition('contains'), true);
+	  this.grid1.markForCheck();
+	}
+  
+	public showAlert(element: ElementRef) {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:203]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 14]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  this.showOverlay = true;
+	  this._overlayId = this.overlayService.attach(element, this.overlaySettings);
+	  this.overlayService.show(this._overlayId);
+	}
+  
+	public hideAlert() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:209]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 15]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  this.showOverlay = false;
+	  this.overlayService.hide(this._overlayId);
+	}
+  
+	private generateRandomNumber(min: number, max: number) {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:214]: set type to `number`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 16]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  return Math.floor(Math.random() * (max - min + 1)) + min;
+	}
+  
+	private ticker() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:218]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 17]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+	  }
+	  this.updateData();
+	  this.manageRace();
+	}
+  
+	private getRandomNumber(min: number, max: number): number {
+	  return Math.round(min + Math.random() * (max - min));
+	}
+  
+	private manageRace() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:235]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 18]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
+  
+	  // show Top 3 players after race has finished
+	  if (this.localData[this.localData.length - 1].TrackProgress === 100) {
+		this.top3 = this.localData.slice(0, 3);
+		this.isFinished = true;
+		this.showAlert(this.finishedAlert);
+	  }
+	}
+  
+	private updateData() {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:259]: set type to `void`

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 19]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
-    // show Top 3 players after race has finished
-    if (this.localData[this.localData.length - 1].TrackProgress === 100) {
-      this.top3 = this.localData.slice(0, 3);
-      this.isFinished = true;
-      this.showAlert(this.finishedAlert);
-    }
+  
+  class CustomTopSpeedSummary {
+  
+	public operate(data?: any[]): IgxSummaryResult[] {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:296]: data should be of type `number[]` as this is on number type column

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts [hunk 20]
```diff
@@ -32,306 +30,315 @@ styleUrls: ['./<%=filePrefix%>.component.scss']
...
-      } else if (position === 0) {
-        elem.Position = 'current';
-      } else {
-        elem.Position = 'up';
-      }
-    });
+  
+  export class CustomBPMSummary {
+  
+	public operate(data?: any[]): IgxSummaryResult[] {
```
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:313]: data should be of type `number[]` as this is on number type column

**Review by wnvko — CHANGES_REQUESTED** [review]
Awesom Grid comments added.
Note - all the files have mixed tabs and spaces for indentation. There should not be any tabs, only spaces.

File: packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts (hunk 10 above)
Comment (wnvko) [inline packages/igx-templates/igx-ts/custom-templates/awesome-grid/files/src/app/__path__/__filePrefix__.component.ts:172]: there is a typo in the method name
```suggestion
	public addSpeedData(athlete: Athlete, minutes?: number): any[] {
```

**Review by wnvko — CHANGES_REQUESTED** [review]

**Review by wnvko — APPROVED** [review]
We should log separate issue to clean up the samples code. There are a lot more not strongly typed items in the code, mixed indentations and so on.

**Comment by valadzhov** [comment]
There are multiple errors when I try to build a project created with the templates:
image

**Review by wnvko — APPROVED** [review]

**Pull request merged**